File: 3. Problem Sets/Week 1/cash/cash.c
//#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    // Ask how many cents the customer is owed
    int cents = get_cents();

    // Calculate the number of quarters to give the customer
    int quarters = calculate_quarters(cents);
    printf("Quarters: %i\n", quarters);
    cents = cents - quarters * 25;

    // Calculate the number of dimes to give the customer
    int dimes = calculate_dimes(cents);
    printf("Dimes: %i\n", dimes);
    cents = cents - dimes * 10;

    // Calculate the number of nickels to give the customer
    int nickels = calculate_nickels(cents);
    printf("Nickels: %i\n", nickels);
    cents = cents - nickels * 5;

    // Calculate the number of pennies to give the customer
    int pennies = calculate_pennies(cents);
    printf("Pennies: %i\n", pennies);
    cents = cents - pennies * 1;

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print total number of coins to give the customer
    printf("Total Coins: %i\n", coins);
}

/**
 * Prompts the user to enter the total of cents owed
*/
int get_cents(void)
{
    int cents;
    do{
        printf("Change owed: ");
        scanf("%i", &cents);
    }
    while(cents < 1);

    return cents;
}

int calculate_quarters(int cents)
{
    return cents / 25;
}

int calculate_dimes(int cents)
{
    return cents / 10;
}

int calculate_nickels(int cents)
{
    return cents / 5;
}

int calculate_pennies(int cents)
{
    return cents / 1;
}

/*
Change owed: 142
Quarters: 5
Dimes: 1
Nickels: 1
Pennies: 2
Total Coins: 9
*/

File: 3. Problem Sets/Week 6/sentimental-cash/cash.py
def main():
    # loop ensures valid user input
    while True:
        try:
            change = float(input("Change owed: $"))
            if change < 0:
                print("Change cannot be negative.")
            else:
                break # break out of loop if input is a non-negative number
        except ValueError:
            print("Invalid Input...")

    change = round(change, 2) * 100 # rounds user input to 2 decimal places & then converts from dollars to cents
    
    quarters = getQuarters(change)
    change -= 25 * quarters

    dimes = getDimes(change)
    change -= 10 * dimes

    nickels = getNickels(change)
    change -= 5 * nickels

    pennies = change # any remaining change will be pennies

    coins = quarters + dimes + nickels + pennies
    print(f"{coins}")


# determines how many quarters to give
def getQuarters(cents):
    return cents // 25 # floor division (int division: rounds down to whole number)


# determines how many dimes to give
def getDimes(cents):
    return cents // 10 # floor division (int division: rounds down to whole number)


# determines how many nickels to give
def getNickels(cents):
    return cents // 5 # floor division (int division: rounds down to whole number)


main() # call to main function

"""
dan@Dan-PC:/mnt/c/Users/Dan/xCS50/3. Problem Sets/Week 6/sentimental-cash$ python cash.py
Change owed: $-1
Change cannot be negative.
Change owed: $test
Invalid Input...
Change owed: $0.41
4
"""